apps/extension/src/approve-origin.ts

```diff
@@ -35,10 +41,19 @@ export const approveOrigin = async ({
       title: tab.title,
     },
   });
+
   if ('error' in res)
     throw errorFromJson(res.error as JsonValue, undefined, ConnectError.from(res));
 
-  connectedSites[urlOrigin] = res.data.attitude;
-  void localExtStorage.set('connectedSites', connectedSites); // TODO: is there a race condition here?
-  return Boolean(connectedSites[urlOrigin]);
+  // TODO: is there a race condition here?
+  // if something has written after our initial read, we'll clobber them
+  void localExtStorage.set('connectedSites', [
```

Comment:
why not await it?

Reply:
we don't need the write to be successful in order to convey the user's response